plugins/targetlocker/inmemory/inmemory.go

```diff
@@ -47,18 +47,34 @@ type lock struct {
 	expiresAt time.Time
 }
 
+func validateRequest(req *request) error {
+	if req == nil {
+		return fmt.Errorf("got nil request")
+	}
+	if req.owner == 0 {
+		return fmt.Errorf("owner cannot be zero")
+	}
+	if len(req.targets) == 0 {
+		return fmt.Errorf("no target specified")
+	}
+	return nil
+}
+
 // broker is the broker of locking requests, and it's the only goroutine with
 // access to the locks map, in accordance with Go's "share memory by
 // communicating" principle.
-func broker(lockRequests, unlockRequests, checkLocksRequests <-chan request, done <-chan struct{}) {
+func broker(lockRequests, unlockRequests, checkLocksRequests <-chan *request, done <-chan struct{}) {
```

Comment:
made this a pointer. The object is modified here, and without a pointer the changes are not reflected at the caller (i.e. the function that writes to the requests channels)